logexporter/cmd/main.go

```diff
@@ -130,9 +130,7 @@ func prepareLogfiles(logDir string) {
 	logfiles := nodeLogs[:]
 
 	switch *cloudProvider {
-	case "gce", "gke":
-		logfiles = append(logfiles, gceLogs...)
-	case "kubemark":
+	case "gce", "gke", "kubemark":
```

Comment:
The problem with this is we're assuming kubemark is running on GCE (which may not necessarily be true). Hence the TODO below. To do this correctly, we must know 2 things:

- are we running kubemark
- what's the real provider

Reply:
If we can pipe that information correctly, we can solve this properly.

Reply:
I'm not sure if I understand.

If we pass --cloud-provider=gce/gke and --enable-hollow-node-logs I think we should fetch hollow node logs in addition to regular kubelet/docker etc logs. We don't assume here that in general kubemark can run only on gce/gke, but we do support fetching logs only in that cas which I believe is fine (if we start running the test on e.g. aws we can implement this later).

If your concern is what should happen when --cloud-provider=kubemark, I think we can simply remove this case as we don't use that (in fact we run this logexporter inside of normal gce cluster, not 'kubemark' cluster).

Reply:
Ok let me clarify. Right now, if we have `cloudProvider == kubemark`, we're also collecting gce logs (which is based on assumption that we're running kubemark on gce (which may not be true)). So it's not a matter of supporting only gce, but a correctness issue (as we shouldn't collect gce logs if kubemark is running on bare-metal for e.g).

> If your concern is what should happen when --cloud-provider=kubemark, I think we can simply remove this case as we don't use that (in fact we run this logexporter inside of normal gce cluster, not 'kubemark' cluster).

If that's the case, let's just remove it. Also can you move the following block outside of the switch-case:

```
 		if *enableHollowNodeLogs {
 			logfiles = append(logfiles, kubemarkLogs...)
 		}
```

Reply:
(assuming I'm understanding correctly)